tests-ng/test_time_config.py

```diff
@@ -0,0 +1,176 @@
+def test_correct_ntp_on_gcp(timedatectl: TimeDateCtl):
+    ntp_hostname = timedatectl.get_ntpserver().hostname
+    assert (
+        ntp_hostname == "metadata.google.internal"
+    ), f"ntp server is invalid. Expected 'metadata.google.internal' got '{ntp_hostname}'."
+
+
+@pytest.mark.flaky(reruns=10, reruns_delay=30, only_rerun="AssertionError")
+@pytest.mark.booted(reason="NTP server configuration is read at runtime")
+@pytest.mark.feature("not azure and not aws and not gcp and not gdch and not qemu")
+def test_ntp(timedatectl: TimeDateCtl):
+    """
+    Validate that NTP is enabled and synchronized on non-hyperscaler platforms.
+    Skips on Azure, AWS and GCP since their metadata-based NTP servers and not reachable from QEMU tests.
+    """
+    # Verify image configuration
+    assert (
+        timedatectl.has_timesync_installed()
+    ), "systemd-timesyncd.service should be present on the image."
+
+    # Check activity and sync when present
+    if timedatectl.is_timesyncd_active():
+        status = timedatectl.get_timesync_status()
+        assert status.ntp, "NTP should be enabled"
+        assert status.ntp_synchronized, "NTP should be synchronized"
+    else:
+        pytest.skip("systemd-timesyncd installed but not active in this environment")
+
+
+@pytest.mark.booted(reason="NTP server configuration is read at runtime")
+@pytest.mark.feature("azure")
+@pytest.mark.hypervisor("microsoft")
+def test_systemd_timesyncd_disabled_on_azure(systemd: Systemd):
+    assert (
+        systemd.is_active("systemd-timesyncd") == False
+    ), f"Chrony instead of systemd-timesyncd should be active on Azure."
+
+
+@pytest.mark.booted(reason="NTP server configuration is read at runtime")
+@pytest.mark.feature("azure and not qemu")
+def test_chrony_on_azure(systemd: Systemd, systemd_detect_virt: Hypervisor):
+    """
+    Test for chrony as active time sync service on Azure.
+    See: https://learn.microsoft.com/en-us/azure/virtual-machines/linux/time-sync#chrony
+    """
+    if systemd_detect_virt == Hypervisor.microsoft:
+        # If real Azure, this will work.
+        assert systemd.is_active("chrony"), f"Chrony should be active on Azure."
+    else:
+        # Azure image tested in QEMU (no Hyper-V clock available -> chrony is disabled and not loaded)
+        units = systemd.list_installed_units()
+        chrony_unit = next(
+            (unit for unit in units if unit.unit == "chrony.service"),
+            None,
+        )
+
+        assert (
+            chrony_unit is not None
+        ), "chrony.service should be present in Azure images."
+        assert chrony_unit.load in (
+            "enabled",
+            "disabled",
+        ), f"unexpected chrony.service state: {chrony_unit.load!r}"
+
+
+@pytest.mark.booted(reason="NTP server configuration is read at runtime")
+@pytest.mark.feature("not azure and not container")
+@pytest.mark.arch("amd64")
+def test_clocksource_x86_64(systemd_detect_virt: Hypervisor, clocksource: str):
```

Comment:
Please use `amd64` in test name for consistency.

Reply:
Fixed it.